### DISCUSSION TITLE
Add APPROX_TOP_COUNT function

### DISCUSSION THREAD

**Discussion by david542542** [duckdb/duckdb-wasm#1711] [Ideas]
BigQuery has a nice feature/function called [`APPROX_TOP_COUNT`](https://cloud.google.com/bigquery/docs/reference/standard-sql/approximate_aggregate_functions#approx_top_count) that returns a struct of the top N values with their approximate count. This is similar to the DuckDB [`SUMMARIZE`](https://duckdb.org/docs/guides/meta/summarize) command, but allows a column (or other arbitrary expression). I think this would be a great function/feature to add if this is possible. Postgres also has `most_common_vals` within [`pg_stats`](https://www.postgresql.org/docs/7.3/planner-stats.html).

**Comment by carlopi** [comment]
Hi! Thanks, looks a cool feature request, can you maybe move this to [main duckdb's repo discussions](https://github.com/duckdb/duckdb/discussions)?

Thanks!

**Comment by soerenwolfers** [comment]
since implemented as `approx_top_k` https://github.com/duckdb/duckdb/pull/12653

**Comment by carlopi** [comment]
Thanks, indeed this will be available in a month or so with duckdb release 1.2
